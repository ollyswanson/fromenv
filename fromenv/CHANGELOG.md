# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [Unreleased]

## [0.1.0](https://github.com/ollyswanson/fromenv/releases/tag/fromenv-v0.1.0) - 2025-08-10

### Added

- Adjust error message based on error count
- Add "path" to error reporting
- Improve diagnostics from unrecognized fields
- requirements

### Fixed

- Update pub struct error message

### Other

- Specify version of fromenv-derive
- Add missing items to manifest
- Use cargo-rdme to interpolate rustdocs
- Add licenses
- Remove `FromEnvError` from public API
- Document usage + motivation
- Include Send + Sync bounds on BoxError
- Rename `ParserResult` to `ParseResult`
- Test error messages + rename ui tests
- Add more trybuild tests
- Scaffold trybuild tests
- Global rename to `FromEnv`
