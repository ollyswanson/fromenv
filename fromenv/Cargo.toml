[package]
name = "fromenv"
version = "0.1.0"
authors = { workspace = true }
description = { workspace = true }
edition = { workspace = true }
keywords = { workspace = true }
license = { workspace = true }
readme = { workspace = true }
repository = { workspace = true }

[dependencies]
fromenv-derive = { path = "../fromenv-derive" }

[dev-dependencies]
insta = "1.43.1"
temp-env = "0.3.6"
trybuild = "1.0.110"
secrecy = "0.10.3"
