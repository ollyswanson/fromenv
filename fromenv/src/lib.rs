mod error;
mod parser;

#[doc(hidden)]
#[path = "private.rs"]
pub mod __private;

pub use error::{FromEnvError, FromEnvErrors};
pub use fromenv_derive::FromEnv;
pub use parser::ParserResult;

type BoxError = Box<dyn std::error::Error + 'static>;
