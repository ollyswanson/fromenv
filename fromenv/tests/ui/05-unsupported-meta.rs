use fromenv::FromEnv;

#[derive(FromEnv)]
pub struct Config {
    #[env(from, invalid)]
    foo: String,
}

fn main() {}
