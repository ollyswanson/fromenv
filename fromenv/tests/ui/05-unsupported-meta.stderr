error: Unknown field: `invalid`
 --> tests/ui/05-unsupported-meta.rs:5:17
  |
5 |     #[env(from, invalid)]
  |                 ^^^^^^^
